pkg/loghttp/query.go

```diff
@@ -55,9 +55,122 @@ func (q *QueryResponse) UnmarshalJSON(data []byte) error {
 	})
 }
 
-// PushRequest models a log stream push
+// PushRequest models a log stream push but is unmarshalled to proto push format.
 type PushRequest struct {
-	Streams []*Stream `json:"streams"`
+	Streams []LogProtoStream `json:"streams"`
```

Comment:
This comes from this PR:
https://github.com/grafana/loki/pull/1145

The only reason why I think this is a pointer is so we don't make a copy of the stream when calling NewStream which receives a ptr to `loghttp.Stream`
https://github.com/grafana/loki/pull/1145/files#diff-71619e1c80a73b34eade235a55d012a0ddbb3375b8d4ac89c1f4fd672145b915R34

With Sandeep's changes, we no longer use that since now we decode from json to `loghttp.PushRequest` to then cast it to `logproto.PushRequest`
https://github.com/grafana/loki/blob/44b0932247909e5d196087b88ec15ae00ad7254d/pkg/util/unmarshal/unmarshal.go#L22

So having that as a ptr or not shouldn't make much difference.